skrf/network.py

```diff
@@ -2963,6 +3063,34 @@ def connect(ntwkA, k, ntwkB, l, num=1):
         ntwkC.renumber(from_ports=from_ports,
                        to_ports=to_ports)
 
+    # if ntwkA and ntwkB are both 2port, and either one has noise, calculate ntwkC's noise
+    if num == 1 and ntwkA.nports == 2 and ntwkB.nports == 2 and (ntwkA.noise is not None or ntwkB.noise is not None):
```

Comment:
wow what a conditional!

Reply:
Haha, yeah! I couldn't think of an easier to way, but it is a little shorter now